nordicsemi/dfu/dfu_transport_ble.py

```diff
@@ -231,33 +231,33 @@ def jump_from_buttonless_mode_to_bootloader(self, buttonless_uuid):
 
     def verify_stable_connection(self):
         """ Verify connection event, and verify that unexpected disconnect
-         events are not received.
+         events are not received. Reconnect if necessary.
 
         Returns:
             True if connected, else False.
 
         """
-        self.conn_handle = self.evt_sync.wait('connected', timeout=30)
-        if self.conn_handle is not None:
-            retries = DFUAdapter.CONNECTION_ATTEMPTS
-            while retries:
-                if self.evt_sync.wait('disconnected', timeout=1) is None:
-                    break
-
-                logger.warning("Received unexpected disconnect event, "
-                               "trying to re-connect to: {}".format(self.target_device_addr))
-                time.sleep(1)
-
-                self.adapter.connect(address=self.target_device_addr_type,
-                                     conn_params=self.conn_params)
-                self.conn_handle = self.evt_sync.wait('connected')
-                retries -= 1
+        logger.info("Checking connection stability")
+        retries = DFUAdapter.CONNECTION_ATTEMPTS
+        while retries:
```

Comment:
IMO is better and more safe to use `for _ in range(retries):` instead of decreasing it every loop, but it's not necessary.